### PR TITLE
[MIG] sale_stock_operating_unit: Migration to 17.0

### PR DESCRIPTION
**Pull request by ghost**
Depends on:

- [x] https://github.com/OCA/operating-unit/pull/667
- [x] https://github.com/OCA/operating-unit/pull/669
- [x] https://github.com/OCA/operating-unit/pull/670
- [x] https://github.com/OCA/operating-unit/pull/671
- [x] https://github.com/OCA/operating-unit/pull/672
- [x] https://github.com/OCA/operating-unit/pull/673

### CODE REVIEW

File: sale_stock_operating_unit/models/stock_move.py [hunk 1]
```diff
@@ -12,7 +12,7 @@ def _get_new_picking_values(self):
         """
         Override to add Operating Units to Picking.
         """
-        values = super(StockMove, self)._get_new_picking_values()
+        values = super()._get_new_picking_values()
```
Comment (nguyenminhchien) [inline sale_stock_operating_unit/models/stock_move.py:15]: in my opinion, it's not necessary to override this function. we have a constraint to make sure the order and its warehouse  having the same value of `operating_unit_id`. So, just let the field `stock_picking.operating_unit_id` computes the value itself.

File: sale_stock_operating_unit/models/stock_warehouse.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+    @api.constrains("operating_unit_id")
+    def _check_existing_so_in_wh(self):
+        for rec in self:
+            sales = self.env["sale.order"].search(
+                [
+                    ("warehouse_id", "=", rec.id),
+                    ("operating_unit_id", "!=", rec.operating_unit_id.id),
+                ]
+            )
```
Comment (nguyenminhchien) [inline sale_stock_operating_unit/models/stock_warehouse.py:21]: should add limit=1

**Review by nguyenminhchien — APPROVED** [review]
Look good overall, just left 2 comments for considering.

File: sale_stock_operating_unit/models/stock_warehouse.py (hunk 2 above)
Comment (ghost) [reply sale_stock_operating_unit/models/stock_warehouse.py:21]: Done 👍

File: sale_stock_operating_unit/models/stock_move.py (hunk 1 above)
Comment (ghost) [reply sale_stock_operating_unit/models/stock_move.py:15]: The operating unit is not mandatory on the warehouse and then you could have a shared warehouse and having pickings from several operating units. In that case, the compute will not work.

**Comment by yvaucher** [comment]
All dependencies have been merge, please refresh the PR by removing the test-requirement change.

**Review by yvaucher — APPROVED** [review]
LGTM

**Review by AaronHForgeFlow — APPROVED** [review]

**Pull request merged**